src/frontend/web_application/src/components/MessageList/components/Message/index.jsx

```diff
@@ -15,48 +34,117 @@ class Message extends Component {
 
   constructor(props) {
     super(props);
-    this.state = { isActive: false };
-    this.handleHover = this.handleHover.bind(this);
-    this.handleBlur = this.handleBlur.bind(this);
+    this.state = {
+      body: '',
+      isExpanded: true,
+      isTooLong: false,
+      isActive: false,
+    };
+    this.handleActiveClick = this.handleActiveClick.bind(this);
+    this.handleExpandClick = this.handleExpandClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.setBodyHeight();
+  }
+
+  setBodyHeight() {
+    const message = this.props.message;
+    const body = message.body;
+    if (body.length > 140) {
+      this.setState({
+        isTooLong: true,
+        isExpanded: false,
+        body: body.substring(0, 140),
+      });
+    } else {
+      this.setState({
+        body,
+      });
+    }
   }
 
-  handleHover() {
-    this.setState({ isActive: true });
+  handleActiveClick() {
+    this.setState(prevState => ({
+      isActive: !prevState.isActive,
+    }));
   }
 
-  handleBlur() {
-    this.setState({ isActive: false });
+  handleExpandClick() {
+    const message = this.props.message;
+    let body = message.body;
```

Comment:
it's almost good, it still uses an eventual old state due to `if (this.state.isExpanded) {`:

```
this.setState((prevState) => {
  return {
    isExpanded: !prevState.isExpanded,
    body: prevState.isExpanded ? body.substring(0, 140) : body,
  };
});
```